postage_stamps.py

```diff
@@ -0,0 +1,272 @@
+#!/usr/bin/env python3.6
+# -*- coding: UTF-8 -*-
+
+import argparse
+import sys
+from copy import copy
+from pathlib import Path
+
+# Todo: Find a way to do this that doesn't involve my user directory
+# User installed packages
+site_pckgs_path = '/global/u1/d/djp81/.local/lib/python3.6/site-packages'
+gcr_catalogs_path = '/global/u1/d/djp81/gcr-catalogs'
+sys.path.extend((site_pckgs_path, gcr_catalogs_path))
+
+import numpy as np
+from astropy.table import Table, vstack
+from tqdm import tqdm
+from astropy.coordinates import SkyCoord
+from astropy import units as u
+
+import GCRCatalogs
+import lsst.afw.geom as afw_geom
+import lsst.daf.persistence as dafPersist
+from lsst.pex.exceptions import LengthError
+
+
+def get_valid_dataids(butler, **kwargs):
+    """Return all dataId values that exist
+
+    Args:
+        butler (Butler): A data access butler
+
+    Returns:
+        A list of dataID values
+    """
+
+    subset = butler.subset('deepDiff_diaSrc')
+    id_list = [dr.dataId for dr in tqdm(subset) if dr.datasetExists()]
+    if not id_list:
+        raise RuntimeError('No valid dataId found.')
+    return id_list
+
+
+def get_truth_catalog():
+    """Load catalog data from "dc2_truth_run1.2_variable_summary"
+
+    Returns:
+       A SourceCatalog object with id, coord_ra, and coord_dec fields
+    """
+
+    # Load truth data
+    truth_gcr = GCRCatalogs.load_catalog('dc2_truth_run1.2_variable_summary')
+    truth_data = truth_gcr.get_quantities(['uniqueId', 'ra', 'dec'])
+
+    truth_table = Table(
+        data=[truth_data['uniqueId'], truth_data['ra'], truth_data['dec']],
+        names=['truth_id', 'truth_ra', 'truth_dec'],
+        dtype=[np.int64, float, float]
+    )
+
+    truth_table['truth_ra'].unit = u.degree
+    truth_table['truth_dec'].unit = u.degree
+    return truth_table
+
+
+def get_diasrc_catalog(butler, data_id):
+    """For a collection of data Ids, return a table of all DIA sources
+
+    Args:
+        butler (Butler): A data access butler
+        data_id  (dict): A dictionary of values uniquely identifying an image
+
+    Returns:
+       An astropy table
+    """
+
+    diasrc_table = Table(
+        names=['src_id', 'src_ra', 'src_dec', 'visit', 'filter',
+               'detector'],
+        dtype=[np.int64, float, float, np.int64, str, 'U10']
+    )
+
+    visit = data_id['visit']
+    filter_ = data_id['filter']
+    detector = data_id['detector']
+    for source in butler.get('deepDiff_diaSrc', dataId=data_id):
+        id_ = source['id']
+        ra = source['coord_ra']
+        dec = source['coord_dec']
+        diasrc_table.add_row([id_, ra, dec, visit, filter_, detector])
+
```

Comment:
As a general rule, try to avoid for loops that build something up.  It's not always possible to do so, but it's a good goal.

I think you can simplify this entire function as:
```
diasrc_cat = butler.get('deepDiff_diaSrc', dataId=data_id).asAstropy
diasrc_table = diasrc_cat['id', 'coord_ra', 'coord_dec']
diasrc_table['visit'] = data_id['visit']
diasrc_table['filter'] = data_id['filter']
diasrc_table['detector'] = data_id['detector']
# Although you don't need the following two columns.  You could just use `coord_ra`, `coord_dec` to do the matching.
diasrc_table['ra'] = np.rad2deg(diasrc_table['coord_ra']) * u.deg
diasrc_table['dec'] = np.rad2deg(diasrc_table['coord_dec']) * u.deg